app/controllers/PushPullNotificationController.scala

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright 2021 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package controllers
+
+import config.Constants
+import connectors.PushPullNotificationConnector
+import controllers.actions.AuthAction
+import javax.inject.Inject
+import play.api.libs.json.Json
+import play.api.mvc.{Action, AnyContent, ControllerComponents}
+import uk.gov.hmrc.http.HttpErrorFunctions
+import uk.gov.hmrc.play.bootstrap.backend.controller.BackendController
+import utils.ResponseHelper
+
+import scala.concurrent.{ExecutionContext, Future}
+
+class PushPullNotificationController @Inject() (cc: ControllerComponents,
+                                                authAction: AuthAction,
+                                                pushPullNotificationConnector: PushPullNotificationConnector)
+                                               (implicit ec: ExecutionContext)
+  extends BackendController(cc)
+    with HttpErrorFunctions
+    with ResponseHelper {
+
+  def getBoxInfo(): Action[AnyContent] = authAction.async {
+    implicit request => {
+      request.headers.get(Constants.XClientIdHeader) match {
+        case Some(clientId) => pushPullNotificationConnector.getBox(clientId).map {
+          response =>
+          response match {
+            case Left(error) if(error.statusCode == NOT_FOUND) => NotFound("No box found for your client id")
+            case Left(_) => InternalServerError("Unexpected Error")
+            case Right(box) => Ok(Json.toJson(box))
+          }
+        }
+        case None =>  Future.successful(BadRequest("Client Id Required"))
```

Comment:
We should use JSON errors in all the responses here

Reply:
So essentially just wrap it in a JsString like `BadRequest(JsString("Client..."))` ?

Reply:
No - the errors from our API need to follow the format from the Developer Hub [Reference Guide](https://developer.service.hmrc.gov.uk/api-documentation/docs/reference-guide#errors) - so they need to have a `code` and a `message` field. See e.g. https://github.com/hmrc/common-transit-convention-traders/blob/master/app/models/response/JsonErrorResponse.scala

Reply:
Or https://github.com/hmrc/common-transit-convention-guarantee-balance/blob/master/app/models/errors/BalanceRequestError.scala